src/router_node.c

```diff
@@ -329,12 +330,33 @@ static bool AMQP_rx_handler(void* context, qd_link_t *link)
     qdr_delivery_t *delivery = qdr_node_delivery_qdr_from_pn(pnd);
     bool       next_delivery = false;
 
-    //
-    // Receive the message into a local representation.
-    //
     qd_message_t   *msg   = qd_message_receive(pnd);
+    
+    int oversize = qd_message_exceeded_max_message_size(msg);
+    if (oversize > 0) {
+        // message is bigger than maxMessageSize
+        if (oversize == 1) {
```

Comment:
Q: what if the message is large and was forwarded to the core, then the oversize condition is hit on a later transfer?  Won't there be a qdr_delivery_t outstanding on the core's qdr_link_t undelivered list?